src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -42,27 +45,23 @@ export class BuilderCanvas extends Vue {
   // ###### Validation API Watchers [
   @Watch('activeFlow', {deep: true, immediate: true})
   async onActiveFlowChanged(newFlow: IFlow): Promise<void> {
-    console.debug('watch/activeFlow:', 'active flow has changed from builder canvas, validating ...')
-    await this.validate_flow({flow: newFlow})
+    this.debounceFlowValidation({newFlow})
   }
 
+  debounceFlowValidation = debounce(function (this: any, {newFlow}: {newFlow: IFlow}) {
+    console.debug('watch/activeFlow:', 'active flow has changed from builder canvas, validating ...')
+    this.validate_flow({flow: newFlow})
+  }, DEBOUNCE_VALIDATION_TIMER)
```

Comment:
Debouncers are not required for the solution.
But they:
- provide a cleaner console
- give maybe a better ux (we don't validate until the user finished making the change)
- maybe improve performance

Reply:
I second performance considerations 👍